### PR TITLE
*: fix the output in changefeed out of order (#1247)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #1247 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In ticdc repo:
git pr https://github.com/pingcap/ticdc/pull/1323
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/ticdc.git pr/1323:release-4.0-5c62e67750c0
```

---

### What problem does this PR solve? 

fix the output in changefeed out of order

### What is changed and how it works?
add a middleware between sorter and sink, make the life cycle of sink managed by processor.

before this pr:
image

after this pr:
image

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- fix the output in changefeed out of order

### CODE REVIEW

File: metrics/grafana/ticdc.json [hunk 1]
```diff
@@ -4402,6 +4402,573 @@
         "x": 0,
         "y": 3
       },
+<<<<<<< HEAD
+=======
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:4406]: need resolve

**Review by leoppro — APPROVED** [review]

**Pull request merged**